File: 4248-count-commas-in-range-ii/count-commas-in-range-ii.cpp
class Solution {
public:
    #define ll long long
    long long countCommas(long long n) {
        ll ans = 0;
        ll start = 1000;
        while (start<=n){
            ans += (n-start+1);
            start *= 1000;
        }
        return ans;
    }
};

File: 4248-count-commas-in-range-ii/count-commas-in-range-ii.py
class Solution:
    def countCommas(self, n: int) -> int:
        commas = 1
        ans = 0
        lower = 1000

        while lower <= n:
            upper = 1000*lower - 1
            if upper > n:
                upper = n
            count = upper - lower + 1
            ans += commas * count
            commas += 1
            lower *= 1000

        return ans 